pkg/controller/v1beta1/inferenceservice/controller.go

```diff
@@ -123,6 +123,16 @@ func (r *InferenceServiceReconciler) Reconcile(ctx context.Context, req ctrl.Req
 		return !utils.Includes(constants.ServiceAnnotationDisallowedList, key)
 	})
 
+	deployConfig, err := v1beta1api.NewDeployConfig(r.Client)
+	if err != nil {
+		return reconcile.Result{}, errors.Wrapf(err, "fails to create DeployConfig")
+	}
+
+	if isvcutils.GetDeploymentMode(annotations, deployConfig) == constants.ModelMeshDeployment {
+		r.Log.Info(string(constants.ModelMeshDeployment), " is set as ", constants.DeploymentMode,
+			", skipping reconcile for inference service: ", "apiVersion", isvc.APIVersion, "isvc", isvc.Name)
+		return ctrl.Result{}, nil
+	}
```

Comment:
Let's maybe move this logic above the `finalizerName := "inferenceservice.finalizers"` line above, as I don't think we need the custom finalizer being added in the case of a ModelMesh deployment.